builtin/providers/azurerm/resource_arm_express_route_circuit.go

```diff
@@ -0,0 +1,239 @@
+)
+
+func resourceArmExpressRouteCircuit() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmExpressRouteCircuitCreate,
+		Read:   resourceArmExpressRouteCircuitRead,
+		Update: resourceArmExpressRouteCircuitCreate,
+		Delete: resourceArmExpressRouteCircuitDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": locationSchema(),
+
+			"service_provider_name": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+			},
+
+			"peering_location": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+			},
+
+			"bandwidth_in_mbps": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+
+			"sku": {
+				Type:     schema.TypeSet,
+				Required: true,
+				MinItems: 1,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"tier": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(network.ExpressRouteCircuitSkuTierStandard),
+								string(network.ExpressRouteCircuitSkuTierPremium),
+							}, true),
+							DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+						},
+
+						"family": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(network.MeteredData),
+								string(network.UnlimitedData),
+							}, true),
+							DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+						},
+					},
+				},
+				Set: resourceArmExpressRouteCircuitSkuHash,
+			},
+
+			"allow_classic_operations": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"service_provider_provisioning_state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"service_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceArmExpressRouteCircuitCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	ercClient := client.expressRouteCircuitClient
+
+	log.Printf("[INFO] preparing arguments for Azure ARM ExpressRouteCircuit creation.")
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	location := d.Get("location").(string)
+	serviceProviderName := d.Get("service_provider_name").(string)
+	peeringLocation := d.Get("peering_location").(string)
+	bandwidthInMbps := int32(d.Get("bandwidth_in_mbps").(int))
+	sku := expandExpressRouteCircuitSku(d.Get("sku").(*schema.Set))
```

Comment:
can we pass in `d` here and have the method take care of the casting, to be consistent with the other resources?